src/View.zig

```diff
@@ -0,0 +1,161 @@
+//! A View is effectively an "oversized" Window that can be written to and rendered in pieces.
+
+const std = @import("std");
+const mem = std.mem;
+
+const View = @This();
+
+const Screen = @import("Screen.zig");
+const Window = @import("Window.zig");
+const Unicode = @import("Unicode.zig");
+const Cell = @import("Cell.zig");
+
+/// View Allocator
+alloc: mem.Allocator,
+/// Underlying Screen
+screen: *Screen,
+/// Underlying Window
+win: *Window,
+
+pub const Config = struct {
+    max_width: usize = 10_000,
+    max_height: usize = 10_000,
+    x_pixel: usize = 0,
+    y_pixel: usize = 0,
+};
+pub fn init(alloc: mem.Allocator, unicode: *const Unicode, config: Config) !View {
+    const screen = try alloc.create(Screen);
+    screen.* = try Screen.init(
+        alloc,
+        .{
+            .cols = config.max_width,
+            .rows = config.max_height,
+            .x_pixel = config.x_pixel,
+            .y_pixel = config.y_pixel,
+        },
+        unicode,
+    );
+    const window = try alloc.create(Window);
+    window.* = .{
+        .x_off = 0,
+        .y_off = 0,
+        .width = config.max_width,
+        .height = config.max_height,
+        .screen = screen,
+    };
+    return .{
+        .alloc = alloc,
+        .screen = screen,
+        .win = window,
+    };
+}
+
+pub fn deinit(self: *View) void {
+    self.alloc.destroy(self.win);
+    self.screen.deinit(self.alloc);
+    self.alloc.destroy(self.screen);
+}
+
+/// Render Config f/ `toWin()`
+pub const RenderConfig = struct {
+    x: usize = 0,
+    y: usize = 0,
+    width: Extent = .fit,
+    height: Extent = .fit,
+
+    pub const Extent = union(enum) {
+        fit,
+        max: usize,
+    };
+};
+/// Render a portion of this View to the provided Window (`win`).
+/// This will return the bounded X (col), Y (row) coordinates based on the rendering.
+pub fn toWin(self: *View, win: *const Window, config: RenderConfig) !struct { usize, usize } {
```

Comment:
Argument `win` should be type `vaxis.Window`. No need to use a const pointer

Reply:
I think this signature should be something like:

`pub fn draw(self: *const View, win: vaxis.Window, x_off: usize, y_off: usize) !void`

The width and height are defined by the window we pass in, we only need to define the origin of where we want to draw.

Reply:
I prefer allowing users to render a smaller part of the View to a Window if they'd like. There's nothing stopping them from doing what you're saying (which is the default), but this offers simple flexibility.